hyperspy/model.py

```diff
@@ -838,6 +840,67 @@ def _model_function(self, param):
         to_return = self.__call__(non_convolved=False, onlyactive=True)
         return to_return
 
+    def get_raw_signal_for_linear_fitting(self):
+        """Gets the current signal, for linear fitting"""
+        to_return = self.signal()[np.where(self.channel_switches)]
+        if self.signal.metadata.Signal.binned is True:
+            to_return = to_return / self.signal.axes_manager[-1].scale
+        return to_return
+
+    def get_signal_for_linear_fitting(self):
+        """
+        Gets the current signal minus fixed model components, for linear fitting
+        -This could use the previous function to reduce typing-
+        """
+        self._set_p0()
+        deactivated_components = []
+        for component in self:
+            if component.active:
+                if not component.free_parameters:
+                    for parameter in component.parameters:
+                        if parameter.twin:
+                            deactivated_components.append(component)
+                            break
+                else:
+                    deactivated_components.append(component)
+        for comp in deactivated_components:
+            comp.active = False
+
+        fixed_model = self._model_function(self.p0)
+
+        for comp in deactivated_components:
+            comp.active = True
+        signal = self.signal()[np.where(self.channel_switches)]
+        signal_to_fit_to = signal - fixed_model
+        if self.signal.metadata.Signal.binned is True:
+            signal_to_fit_to = signal_to_fit_to / self.signal.axes_manager[-1].scale
+        return signal_to_fit_to
+
+    def check_all_active_components_are_linear(self):
+        """Loops through active components checking all parameters are linear"""
+        for comp in self:
+            if comp.active:
+                for parameter in comp.parameters:
+                    if parameter.free:
+                        if not parameter.is_linear:
+                            return False
+        return True
+
+    def get_nonlinear_components(self):
```

Comment:
This could be a property e.g. ``nonlinear_components``. Also why not a ``linear_components`` property? I would be useful for users to know what's making their model nonlinear.